toqito/rand/random_psd_operator.py

```diff
@@ -0,0 +1,59 @@
+"Generates a random positive semi-definite operator."
+
+import numpy as np
+
+def random_psd_operator(dim: int) -> np.ndarray:
+    r''' Generate a random positive semi definite operator.
+
+    A positive semi-definite operator is a Hermitian operator that has only real and non-negative eigenvalues. 
+
+    This function generates a random positive semi-definite operator by constructing a symmetric matrix, 
+    based on the fact that a symmetric matrix can have its eigen values as real numbers. First it generates a random
+    density matrix and then creates a symmetry matrix from it. Then the eigen values are made to their absolute values.
+    Then the positive semi definite operator is constructed by taking the dot product of the absolute eigen value matrix,
+    eigen vectors and eigen value transpose. Finally it is normalized to make its trace as 1. The resultant of the function will be a numpy ndarray 
+    which is positive semi-definite and unitary.
+
+    Example
+    ===========================
+    Generate a positive semi-definite operator of dimension 3.
+
+    >>> dim = 3
```

Comment:
The example should showcase the use of the function. Not a redefinition of the whole thing.